com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AnimationSetBuilder.java

```diff
@@ -185,9 +187,36 @@ static void loadModelAnimations(ArrayList<ModelImporter.Bone> bones, AnimationSe
         ModelUtil.unloadScene(scene);
     }
 
-    static public void buildAnimations(List<String> paths, ArrayList<ModelImporter.Bone> bones, List<InputStream> streams, List<String> parentIds,
+    public static class ResourceDataResolver implements ModelImporter.DataResolver
+    {
+        Project project;
+
+        public ResourceDataResolver(Project project) {
+            this.project = project;
+        }
+
+        public byte[] getData(String path, String uri) {
+            File file = new File(path);
+            File bufferFile = new File(file.getParentFile(), uri);
+            IResource resource = project.getResource(bufferFile.getPath());
+            if (resource == null)
+            {
+                System.out.printf("Failed to find data for %s\n", bufferFile.getPath());
+                return null;
+            }
+            try {
+                return resource.getContent();
+            } catch (IOException e) {
+                return null; // Actual errors are reported by ModeulUtil.loadScene
```

Comment:
We keep the api simple, and we double check that all buffers are resolved at the end of the load.